internal/retryhttp/client.go

```diff
@@ -181,32 +180,28 @@ func transport() http.RoundTripper {
 // status codes differently.
 func retryPolicy(ctx context.Context, resp *http.Response, err error) (bool, error) {
 	// do not retry on context.Canceled or context.DeadlineExceeded
-	if ctx.Err() != nil {
+	if ctx.Err() != nil || err == nil {
```

Comment:
```suggestion
	if ctx.Err() != nil {
```

I was wrong about this, status codes over 2xx do not equal a non-nil error.